src/parser/bmp/error.rs

```diff
@@ -31,14 +37,46 @@ impl Display for ParserBmpError {
 
 impl Error for ParserBmpError {}
 
-impl std::convert::From<std::io::Error> for ParserBmpError {
+// TODO: These conversions make the error difficult to debug as they drop all of the error context
```

Comment:
We could use [`thiserror`](https://docs.rs/thiserror/latest/thiserror/) crate for `ParserBmpError` or other errors. We could leave that for another PR if you prefer.

Reply:
I think that would be a good idea. A separate PR may be better since any changes to the main error types will likely require widespread changes to the parser.